src/main/java/ch/kalunight/zoe/service/InfoPanelRefresher.java

```diff
@@ -1,165 +1,430 @@
+      }
+    }
+
+    infoPanelMessages.clear();
+    infoPanelMessages.addAll(InfoChannelRepository.getInfoPanelMessages(server.serv_guildId));
+
+    for(int i = 0; i < infoPanels.size(); i++) {
+      DTO.InfoPanelMessage infoPanel = infoPanelMessages.get(i);
+        infochannel.retrieveMessageById(infoPanel.infopanel_messageId).complete().editMessage(infoPanels.get(i)).queue();
+    }
+  }
+
+  private List<DTO.GameInfoCard> refreshGameCardStatus() throws SQLException {
+    List<DTO.GameInfoCard> gameInfoCards = GameInfoCardRepository.getGameInfoCards(server.serv_guildId);
+
+    for(DTO.GameInfoCard gameInfoCard : gameInfoCards) {
+      if(gameInfoCard.gamecard_status == GameInfoCardStatus.IN_WAIT_OF_MATCH_ENDING) {
+        if(gameInfoCard.gamecard_fk_currentgame == 0) {
+          GameInfoCardRepository.updateGameInfoCardStatusWithId(
+              gameInfoCard.gamecard_id, GameInfoCardStatus.IN_WAIT_OF_DELETING);
+          gameInfoCard.gamecard_status = GameInfoCardStatus.IN_WAIT_OF_DELETING;
+        }
+      }
+    }
+    return gameInfoCards;
+  }
+
+  private void createMissingInfoCard() throws SQLException {
+    List<DTO.CurrentGameInfo> currentGamesWithoutCard = 
+        CurrentGameInfoRepository.getCurrentGameWithoutLinkWithGameCardAndWithGuildId(server.serv_guildId);
+
+    List<DTO.CurrentGameInfo> alreadyGeneratedGames = new ArrayList<>();
+
+    DTO.InfoChannel infochannel = InfoChannelRepository.getInfoChannel(server.serv_guildId);
+
+    for(DTO.CurrentGameInfo currentGame : currentGamesWithoutCard) {
+      boolean alreadyGenerated = false;
+      for(DTO.CurrentGameInfo alreadyGeneratedGame : alreadyGeneratedGames) {
+        if(alreadyGeneratedGame.currentgame_currentgame.getGameId() == currentGame.currentgame_currentgame.getGameId()) {
+          alreadyGenerated = true;
+        }
+      }
+
+
+      if(!alreadyGenerated) {
+        GameInfoCardRepository.createGameCards(
+            infochannel.infoChannel_id, currentGame.currentgame_id, GameInfoCardStatus.IN_CREATION);
+        alreadyGeneratedGames.add(currentGame);
+      }
+
+
+      linkAccountWithGameCard(currentGame);
+    }
+  }
+
+  private void linkAccountWithGameCard(DTO.CurrentGameInfo currentGame) throws SQLException {
+    DTO.GameInfoCard gameCard = GameInfoCardRepository.
+        getGameInfoCardsWithCurrentGameId(server.serv_guildId, currentGame.currentgame_id);
+
+    List<DTO.LeagueAccount> leaguesAccountInTheGame = 
+        LeagueAccountRepository.getLeaguesAccountsWithCurrentGameId(currentGame.currentgame_id);
+
+    for(DTO.LeagueAccount leagueAccount : leaguesAccountInTheGame) {
+      LeagueAccountRepository.updateAccountGameCardWithAccountId(leagueAccount.leagueAccount_id, gameCard.gamecard_id);
+    }
+  }
+
+  private void refreshAllLeagueAccountCurrentGamesAndDeleteOlderInfoCard(List<DTO.Player> playersDTO) throws SQLException {

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'refreshAllLeagueAccountCurrentGamesAndDeleteOlderInfoCard(List)' has an NPath complexity of 802](https://app.codacy.com/manual/KaluNight/Zoe-Discord-Bot/pullRequest?prid=4739924)